homeassistant/components/rotel/media_player.py

```diff
@@ -0,0 +1,493 @@
+"""
+Rotel platform.
+
+Although only the RSP-1570 is supported at the moment, the
+low level library could easily be updated to support other
+products of a similar vintage.
+"""
+import asyncio
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.media_player import (
+    DOMAIN, PLATFORM_SCHEMA, MediaPlayerDevice)
+from homeassistant.components.media_player.const import (
+    SUPPORT_SELECT_SOURCE, SUPPORT_TURN_OFF, SUPPORT_TURN_ON,
+    SUPPORT_VOLUME_MUTE, SUPPORT_VOLUME_SET, SUPPORT_VOLUME_STEP)
+from homeassistant.const import (
+    ATTR_ENTITY_ID, CONF_DEVICE, CONF_NAME, EVENT_HOMEASSISTANT_STOP,
+    STATE_OFF, STATE_ON)
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = 'Rotel RSP-1570'
+
+SUPPORT_ROTEL_RSP1570 = \
+    SUPPORT_VOLUME_SET | SUPPORT_VOLUME_STEP | SUPPORT_VOLUME_MUTE | \
+    SUPPORT_TURN_ON | SUPPORT_TURN_OFF | SUPPORT_SELECT_SOURCE
+
+ROTEL_RSP1570_SOURCES = {
+    ' CD': 'SOURCE_CD',
+    'TUNER': 'SOURCE_TUNER',
+    'TAPE': 'SOURCE_TAPE',
+    'VIDEO 1': 'SOURCE_VIDEO_1',
+    'VIDEO 2': 'SOURCE_VIDEO_2',
+    'VIDEO 3': 'SOURCE_VIDEO_3',
+    'VIDEO 4': 'SOURCE_VIDEO_4',
+    'VIDEO 5': 'SOURCE_VIDEO_5',
+    'MULTI': 'SOURCE_MULTI_INPUT',
+}
+
+CONF_SOURCE_ALIASES = "source_aliases"
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_DEVICE): cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_SOURCE_ALIASES): vol.Schema(
+        {vol.Any(*ROTEL_RSP1570_SOURCES.keys()): vol.Any(str, None)}),
+})
+
+ATTR_SOURCE_NAME = "source_name"
+ATTR_VOLUME = "volume"
+ATTR_MUTE_ON = "mute_on"
+ATTR_PARTY_MODE_ON = "party_mode_on"
+ATTR_INFO = "info"
+ATTR_ICONS = "icons"
+ATTR_SPEAKER_ICONS = "speaker_icons"
+ATTR_STATE_ICONS = "state_icons"
+ATTR_INPUT_ICONS = "input_icons"
+ATTR_SOUND_MODE_ICONS = "sound_mode_icons"
+ATTR_MISC_ICONS = "misc_icons"
+ATTR_TRIGGERS = "triggers"
+
+ATTR_COMMAND_NAME = "command_name"
+SERVICE_SEND_COMMAND = "rotel_send_command"
+SERVICE_SEND_COMMAND_SCHEMA = vol.Schema({
+    vol.Required(ATTR_ENTITY_ID): cv.comp_entity_ids,
+    vol.Required(ATTR_COMMAND_NAME): cv.string,
+})
+SERVICE_RECONNECT = "rotel_reconnect"
+SERVICE_RECONNECT_SCHEMA = vol.Schema({
+    vol.Required(ATTR_ENTITY_ID): cv.comp_entity_ids,
+})
+
+
+async def async_setup_platform(
+        hass,
+        config,
+        async_add_entities,
+        discovery_info=None):
+    """Set up the rsp1570serial platform."""
+    # pylint: disable=unused-argument
+
+    device = RotelMediaPlayer(
```

Comment:
This is a `entity` not device (I know, the base class name is confusing -- it's a legacy issue)

```suggestion
    entity = RotelMediaPlayer(
```